library/tests/src/test_arithmetic.rs

```diff
@@ -0,0 +1,401 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::test_expression;
+use qsc::interpret::Value;
+
+// Tests for Microsoft.Quantum.Arithmetic namespace
+
+#[test]
+fn check_apply_xor_in_place() {
+    test_expression(
+        {
+            "{
+            use a = Qubit[3];
+            mutable result = [];
+            within {
+                Microsoft.Quantum.Arithmetic.ApplyXorInPlace(3, a);
+            }
+            apply {
+                set result = [M(a[0]),M(a[1]),M(a[2])];
+            }
+            return result;
+        }"
+        },
+        &Value::Array(vec![Value::RESULT_ONE, Value::RESULT_ONE, Value::RESULT_ZERO].into()),
+    );
+}
+
+#[test]
+fn check_apply_xor_in_place_l() {
+    test_expression(
+        {
+            "{
+            open Microsoft.Quantum.Arithmetic;
+            use q = Qubit[100];
+            ApplyXorInPlaceL(953L <<< 50, q);
+            let result = MeasureInteger(q[50...]);
+            ResetAll(q);
+            result
+        }"
+        },
+        &Value::Int(953),
+    );
+}
+
+#[test]
+fn check_measure_integer() {
+    test_expression(
+        {
+            "{
+                open Microsoft.Quantum.Arithmetic;
+                use q = Qubit[16];
+                ApplyXorInPlace(45967, q);
+                let result = MeasureInteger(q);
+                ResetAll(q);
+                return result;
+            }"
+        },
+        &Value::Int(45967),
+    );
+}
+
+#[test]
+fn check_add_i_nc() {
```

Comment:
I suggest to exhaustively test the adders for all possible input value combinations up to some bit width. There are a lot of corner cases to consider.

Reply:
I also need to add tests for controlled versions.